runtime/buffer.cc

```diff
@@ -345,18 +345,23 @@ bool can_slice_with(const raw_buffer& buf, const raw_buffer& other_buf) {
 }
 #endif
 
+bool is_contiguous_slice(const raw_buffer* const* bufs, std::size_t size, int d) {
+  for (std::size_t n = 0; n < size; n++) {
```

Comment:
Since this will mostly get unrolled, I wonder if we'd get better codegen (ie no conditional branches) by writing as

bool ok = true;
...
ok &= bufs[n]->dim(d).stride() == static_cast<index_t>(bufs[n]->elem_size);